tools/spark-reader-v1/src/main/java/com/vesoft/nebula/reader/ScanEdgeIterator.java

```diff
@@ -0,0 +1,93 @@
+import org.slf4j.LoggerFactory;
+
+public class ScanEdgeIterator extends AbstractNebulaIterator {
+
+    private Logger logger = LoggerFactory.getLogger(ScanEdgeIterator.class);
+
+    private Iterator<ScanEdgeResponse> responseIterator;
+
+    public ScanEdgeIterator(ConnectInfo connectInfo, Partition split,
+                            ScanInfo scanInfo, Map<String, Integer> propIndexMap) {
+        super(connectInfo, split, scanInfo, propIndexMap);
+        processor = new ScanEdgeProcessor(metaClient);
+    }
+
+    @Override
+    public boolean hasNext() {
+        if (dataIterator == null && responseIterator == null && !scanPartIterator.hasNext()) {
+            return false;
+        }
+
+        while (dataIterator == null || !dataIterator.hasNext()) {
+            if (responseIterator == null || !responseIterator.hasNext()) {
+                if (scanPartIterator.hasNext()) {
+                    try {
+                        responseIterator = storageClient.scanEdge(connectInfo.getSpaceName(),
+                                scanPartIterator.next(), returnCols,
+                                false, 1000, 0L, Long.MAX_VALUE);
+                    } catch (IOException e) {
+                        logger.error(e.getMessage(), e);
+                    }
+                    continue;
+                }
+                break;
+            } else if (responseIterator.hasNext()) {
+                ScanEdgeResponse next = responseIterator.next();
+                if (next != null) {
+                    Result processResult = processor.process(connectInfo.getSpaceName(), next);
+                    dataIterator = process(processResult);
+                }
+                continue;
+            }
+        }
+        if (dataIterator == null) {
+            return false;
+        }
+        return dataIterator.hasNext();
+    }
+
+    @Override
+    protected Iterator<String[]> process(Result result) {
+        List<String[]> resultValues = new ArrayList<>();
+
+        Map<String, List<com.vesoft.nebula.data.Row>> dataMap = result.getRows();
+        for (Map.Entry<String, List<com.vesoft.nebula.data.Row>> dataEntry : dataMap.entrySet()) {
+            String labelName = dataEntry.getKey();
+            List<Integer> propIndexs = labelPropIndexMap.get(labelName);
+            List<com.vesoft.nebula.data.Row> rows = dataEntry.getValue();
+            for (com.vesoft.nebula.data.Row row : rows) {
+                Iterator<Integer> nameIndexIterator = propIndexs.iterator();
+                String[] fields = new String[propSize + 3];
+                fields[0] = String.valueOf(row.getDefaultProperties()[0].getValue());
+                fields[1] = String.valueOf(row.getDefaultProperties()[2].getValue());
+                Property[] properties = row.getProperties();
+                for (int i = 0; i < properties.length; i++) {
+                    fields[nameIndexIterator.next()] = String.valueOf(properties[i].getValue());
```

Comment:
nameIndexIterator.next() coudle be 0 or 1, is it ok to overwrite the values of fields[0] and fields[1]?

Reply:
the nameIndexIterator only contain the vertex or edge prop, vertex has default value id, edge has default value src and dst. vertex nameIndexiterator.next() could not be 0, edge nameIndexiterator.next() could not be 0 or 1.

the nameIndexIterator's data from labelPropIndexMap, when labelPropIndexMap init, it only add prop and it's index,it could not add default value and it index
``` java
for (Map.Entry<String, List<String>> colEntry : returnCols.entrySet()) {
    List<Integer> colIndexs = new ArrayList<>();
    List<String> propNames = colEntry.getValue();
    for (String propName : propNames) {
        colIndexs.add(propIndexMap.get(propName));
    }
    labelPropIndexMap.put(colEntry.getKey(), colIndexs);
}
```

Reply:
I see. the code guarantees vertex id and edge src、dst being at the head of the list fields., so prop's index would not be 0 or 1.
![image](https://user-images.githubusercontent.com/16240361/92084794-7cd28600-edfa-11ea-828f-7a5b2740bd41.png)